File: src/tf_turtle/src/turtle_spawn.cpp
#include "ros/ros.h"
#include "turtlesim/Spawn.h"

int main(int argc, char **argv) {
    setlocale(LC_ALL, "");
    ros::init(argc, argv, "spawn_turtle");
    ros::NodeHandle nh;

    ros::ServiceClient client = nh.serviceClient<turtlesim::Spawn>("/spawn"); // 创建客户端
    ros::service::waitForService("/spawn");                                   // 等待服务启动

    turtlesim::Spawn spawn;
    spawn.request.name = "turtle2";
    spawn.request.x = 1.0;
    spawn.request.y = 1.0;
    spawn.request.theta = 1.571;

    if (client.call(spawn)) {
        ROS_INFO("Turtle spawned successfully!");
    } else {
        ROS_ERROR("Failed to spawn turtle.");
    }

    return 0;
}

File: src/tf_turtle/scripts/turtle_spawn.py
import rospy
from turtlesim.srv import Spawn, SpawnRequest, SpawnResponse


def main():
    rospy.init_node("spawn_turtle")

    client = rospy.ServiceProxy("/spawn", Spawn)
    client.wait_for_service()

    request = SpawnRequest()
    request.name = "turtle2"
    request.x = 1.0
    request.y = 1.0
    request.theta = 1.571

    if client.call(request):
        rospy.loginfo("Turtle spawned successfully!")
    else:
        rospy.logerr("Failed to spawn turtle.")


if __name__ == "__main__":
    main()
